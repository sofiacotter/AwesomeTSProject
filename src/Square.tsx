import React, {useEffect} from 'react';
import {Pressable, Text} from 'react-native';
import Animated, {
  useAnimatedStyle,
  useSharedValue,
  withRepeat,
  withSpring,
  withTiming,
} from 'react-native-reanimated';
import styles from './stylesheet';

/* Function components: Componentes que apenas contém método render()
e não mantêm estado próprio. Em vez de serem definidos como uma classe
que estende de React.Component, são funções que incluem props como input
e retornam o que deve ser renderizado */

/*É importante fazer assign de keys em listas, para definir a sua ordem
e as operações e remover, adicionar, substituir... as keys não têm de ser
únicas globalmnte, mas sim entre componentes e os seus "irmãos".
<li key={this.props.id}> */

/* Variáveis imutáveis: não substituir as variáveis diretamente, mas sim
por uma cópia das mesmas, alterada */

interface SquareProps {
  isWin: boolean | null;
  value: null | string;
  onClick: () => void;
}

const Square = ({isWin, value, onClick}: SquareProps): JSX.Element => {
  //const progress = useSharedValue(1);
  const scale = useSharedValue(2);

  const animatedStyles = useAnimatedStyle(() => {
    return {
      //opacity: progress.value,
      transform: [{scale: scale.value}],
    };
  }, []);

  const playAnimation = (): void => {
    scale.value = withRepeat(withSpring(1), 2, true);
  };
  const combPress = (): void => {
    playAnimation();
    onClick();
  };

  //useEffect(() => {}, []);

  if (isWin) {
    return (
      //botão verde
      <Pressable style={styles.winsquare} onPress={combPress}>
        <Animated.Text style={(styles.text, animatedStyles)}>
          {value}
        </Animated.Text>
      </Pressable>
    );
  }
  return (
    //botão vermelho
    <Pressable style={styles.square} onPress={combPress}>
      <Animated.Text style={(styles.text, animatedStyles)}>
        {value}
      </Animated.Text>
    </Pressable>
  );
};

export default Square;
