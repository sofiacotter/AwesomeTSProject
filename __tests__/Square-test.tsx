import React from 'react';
import renderer from 'react-test-renderer';
import Square from '../src/Square';
import {fireEvent, render, within} from '@testing-library/react-native';
describe('user related actions', () => {
  //TESTAR A RENDERIZAÇÃO DO QUADRADO
  test('Square Test', () => {
    const tree = renderer
      .create(
        <Square
          isWin={false}
          value={'X'}
          onClick={function (): void {
            throw new Error('Function not implemented.');
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  //TESTAR AS CORES A MUDAREM NO QUADRADO (isWin=true)
  //TESTAR DE O BOTÃO SÓ TEM OS VALORES NULL, 'X' OU 'O'
  /*
  it('Square should have particular values', () => {
    const squareTest = render(
      <Square
        isWin={true}
        value={'X'}
        onClick={function (): void {
          console.log('OnClick Square!');
        }}
      />,
    );
    const squareButton = within(squareTest.value).in
    fireEvent.press(screen.getByText('Print Username'));
  });
  */
});
