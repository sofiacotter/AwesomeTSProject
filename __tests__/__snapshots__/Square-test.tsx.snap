// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square Test 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f5fffa",
      "borderColor": "#b0e0e6",
      "borderWidth": 3,
      "flex": 1,
      "fontWeight": "bold",
      "justifyContent": "center",
      "margin": 2,
      "padding": "5%",
      "textAlign": "center",
    }
  }
>
  <Text
    style={
      Object {
        "transform": Array [
          Object {
            "scale": 2,
          },
        ],
      }
    }
  >
    X
  </Text>
</View>
`;

exports[`user related actions Square Test 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f5fffa",
      "borderColor": "#b0e0e6",
      "borderWidth": 3,
      "flex": 1,
      "fontWeight": "bold",
      "justifyContent": "center",
      "margin": 2,
      "padding": "5%",
      "textAlign": "center",
    }
  }
>
  <Text
    style={
      Object {
        "transform": Array [
          Object {
            "scale": 2,
          },
        ],
      }
    }
  >
    X
  </Text>
</View>
`;
